src/main/java/org/opensearch/security/filter/SecurityRestFilter.java

```diff
@@ -210,4 +253,30 @@ public void onWhitelistingSettingChanged(WhitelistingSettings whitelistingSettin
     public void onAllowlistingSettingChanged(AllowlistingSettings allowlistingSettings) {
         this.allowlistingSettings = allowlistingSettings;
     }
+
+    /**
+     * Determines if the request's path is a match for the configured handler path.
+     *
+     * @param requestPath The path from the {@link NamedRoute}
+     * @param handlerPath The path from the {@link RestHandler.Route}
+     * @return true if the request path matches the route
+     */
+    private boolean restPathMatches(String requestPath, String handlerPath) {
```

Comment:
Not sure what this function works on it looks like basic string comparision, needs unit tests which can clarify